File: Concurrency.java
/**
 * Write a short Java program that uses threads in the following way. There will be one shared 
 * array that is initialized at the start of the program to hold the integers from 0 to 99. 
 * There is also a shared global integer index that starts at 0. You will start two identical 
 * threads, one named A and one named B. Each one will use the synchronized keyword to control 
 * access to the array and the index. Each thread loops, and the synchronized portion checks 
 * if the array index is still in bounds. If it is in bounds, the thread prints its name followed 
 * by the array value, then sets the array value to -1, and finally increments the index. 
 * Afterward, the thread puts itself to sleep with this.sleep(1). Once the index is out of bounds, 
 * the loop/thread exits/terminates. The main method should use the join() method of the threads 
 * to wait for them to finish. Then, at the end of the main method, use a loop to compute the sum 
 * of the array values and print that sum to the console (should be -100). You can define your thread 
 * class by extending Thread with a static class defined inside of the file below.
 * 
 * It is important to note that concurrent programs often exhibit non-deterministic behavior, and
 * that should be the case with this program. However, that means that there is no autograding for
 * this coding problem. Example possible output is in the README, but human inspection will determine
 * whether your code is actually correct.
 */

public class Concurrency  {

    // TODO

}

File: concurrency.py
import threading
import time
import os

"""
    Write a Python program that repeatedly takes console input from the user.
    If the user provides an input of "quit" then the program exits. For any
    other input, the program spawns a new threading.Thread that will repeat
    the given input as output to the console forever, with a 2 second pause
    between each output. Use time.sleep for this pause. The program will 
    continue to accept user input, and therefore there could be multiple
    threads periodically outputting their given string to the console.

    There is no autograding for this coding problem. 
    Human inspection will determine whether your code 
    is actually correct.
"""

PAUSE_BETWEEN = 2 # 2 seconds

# TODO: Define thread function here

if __name__ == "__main__":
    quit = False
    while not quit:
        value = input()
        if value == "quit":
            quit = True

        # TODO: Launch threads in this loop

    # Exits main thread and any other threads that are running
    os._exit(0)